### DISCUSSION TITLE
pypyr.steps.debug, Makes more sense if it only logged for loglevel < 20 ?

### DISCUSSION THREAD

**Discussion by fabiilo** [pypyr/pypyr#349] [Q&A]
Big fan of the function, easy way to drop context to logs.

However, it does feel like more of a "debug" step?
Hence, would it make more sense to have it being run if loglevel < 20 ? 
Rather than with the rest of the info logs.

**Comment by yaythomas** [comment]
sorry it took a while to reply @fabiilo! notification got buried under some other notifications and I missed it 😬 

This is the result of a bit of a legacy with the debug levels. 

Initially, pypyr was very verbose, defaulting to INFO level with pretty chatty output. Later, when some user feedback suggested that LESS output would be less confusing, I added NOTIFY level - which is the default today, and only outputs when specifically asked by a step.

The DEBUG level  is very verbose - to the point where it can take an annoying while to find things, and for me at least, having `pypyr.steps.debug` to INFO at the very least means I can find the output quicker. Generally when I use pypyr.steps.debug I'm more interested in the specific shape of the context at that point, and NOT in the full debug logs that include "x started", "x ended" for every method in the call chain...

 But yes, it does feel strange that something called "debug" is outputting to the INFO level, and not DEBUG!

In hindsight, it would have been better to add a new log level at the bottom for TRACE instead (to be the equivalent of the current DEBUG), and then DEBUG would be today's INFO and INFO would be today's NOTIFY - i.e to bump all the levels down one and then we wouldn't need the custom NOTIFY level as default. Things always seem a lot more obvious when looking back... 😅
